tests/detect_incomplete_text.rs

```diff
@@ -0,0 +1,74 @@
+#![cfg(feature = "experimental-reader-writer")]
+
+use crate::ion_tests::{DataStraw, ELEMENT_GLOBAL_SKIP_LIST};
+use ion_rs::{AnyEncoding, ElementReader, IonError, IonResult, IonStream, Reader};
+use std::collections::HashSet;
+use std::fs;
+use std::io::BufReader;
+use std::iter::Iterator;
+use std::sync::LazyLock;
+use test_generator::test_resources;
+
+mod ion_tests;
```

Comment:
🗺️ This file was moved and modified, so GitHub is showing it as brand new (with the corresponding old file being 'deleted' rather than moved.) The logic here is the same, with two small changes:
1. The processes of canonicalizing the file names for comparison to the skip list has been moved behind [`LazyLock`](https://doc.rust-lang.org/std/sync/struct.LazyLock.html)s so the work only happens once on first use.
2. We now also test the `good` files in `testdata_1_0` in addition to `testdata_1_1`.